File: pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <ctype.h>

void caesar (string p, int k);

int const BEFORELOWERA = 96; //символ в ASCII, который находится перед маленькой а
int const BEFOREUPPERA = 64; //перед большой А

int main (int argc, char* argv[]) {
    if (argc != 2) { //если не был введен ключ, то получаем error
        printf ("ERROR!!!\n");
        return 1;
    }
    
    int k = atoi(argv[1]); //переводим чаровский массив argv[1] в int
    printf ("plaintext: ");
    string p = GetString();
    printf ("ciphertext: ");
    caesar(p, k);
    return 0;
}

void caesar (string p, int k) {
    for (int i = 0; p[i] != '\0'; i++) {
        if ((isalpha(p[i])) && (islower(p[i]))) { //проходим по маленьким буквам
            int temp = (p[i] + k); 
                while (temp > 'z') //цикл для перехода ключа в начало алфавита до тех пор, пока он не будет больше z в ASCII таблице
                    temp = temp - 'z' + BEFORELOWERA;
                printf ("%c", temp);
        }
        else if ((isalpha(p[i])) && (isupper(p[i]))) { //проходим по большим буквам
            int temp = (p[i] + k); 
                while (temp > 'Z') //тоже самое, но для верхнего регистра
                    temp = temp - 'Z' + BEFOREUPPERA;
                printf ("%c", temp);
        }
        else 
            printf ("%c", p[i]); //выводим все остальные символы
    }
    printf ("\n");
}

File: pset6/caesar.py
import cs50
from sys import argv

def main():
    if len(argv) != 2:
        print ("You screwed up!")
        exit(1)
    
    if str.isnumeric(argv[1]) == False:
        print ("please, put only number!")
        exit(2)
    
    key = int(argv[1])
    
    print ("plaintext: ", end="")
    inputText = cs50.get_string()
    print ("ciphertext: ", end="")
    caesar(inputText, key)
    exit(0)
    
def caesar(text, key):
    for i in range(len(text)):
        if str.isalpha(text[i]) and str.islower(text[i]):
            temp = ord(text[i]) + key
            while temp > ord('z'):
                temp = temp - ord('z') + 96
            print (chr(temp), end="")
        elif str.isalpha(text[i]) and str.isupper(text[i]):
            temp = ord(text[i]) + key
            while temp > ord('Z'):
                temp = temp - ord('Z') + 64
            print (chr(temp), end="")
        else:
            print (text[i], end="")
  
if __name__ == "__main__":
    main()
            